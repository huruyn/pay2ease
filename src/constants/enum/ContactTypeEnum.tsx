// eslint-disable-next-line import/prefer-default-export
export enum ContactTypeEnum {
  PERSONAL = 'personal',
  BUSINESS = 'business',
}
