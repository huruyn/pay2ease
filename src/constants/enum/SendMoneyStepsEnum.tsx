// eslint-disable-next-line import/prefer-default-export
export enum SendMoneyStepsEnum {
  TRANSACTION_START = 'transaction-start',
  TRANSACTION_DETAILS = 'transaction-details',
  TRANSACTION_SECURITY_QUESTIONS = 'transaction-security-questions',
  TRANSACTION_STATUS = 'transaction-status',
  TRANSACTION_COMPLETE = 'transaction-complete',
}
