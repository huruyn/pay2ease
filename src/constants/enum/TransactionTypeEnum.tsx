// eslint-disable-next-line import/prefer-default-export
export enum TransactionTypeEnum {
  SEND = 'send',
  REQUEST = 'request',
}
