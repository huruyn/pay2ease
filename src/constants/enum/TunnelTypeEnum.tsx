// eslint-disable-next-line import/prefer-default-export
export enum TunnelTypeEnum {
  ZIPPY_CASH = 'Zippy.Cash',
  INTERAC_E_TRANSFER = 'InteracÂ® e-Transfer',
}
