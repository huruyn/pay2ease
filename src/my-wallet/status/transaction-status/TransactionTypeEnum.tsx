// eslint-disable-next-line import/prefer-default-export
export enum TransactionTypeEnum {
  SENT = 'sent',
  RECEIVED = 'received',
  REQUESTED = 'requested',
}
