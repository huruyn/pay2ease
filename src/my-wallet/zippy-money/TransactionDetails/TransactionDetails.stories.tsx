import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import TransactionDetails from './TransactionDetails';
import { TransactionTypeEnum } from '../../../constants/enum/TransactionTypeEnum';

export default {
  title: 'my-wallet/zippy-money/TransactionDetails',
  component: TransactionDetails,
} as ComponentMeta<typeof TransactionDetails>;

const Template: ComponentStory<typeof TransactionDetails> = () => (
  <BrowserRouter>
    <TransactionDetails transactionType={TransactionTypeEnum.SEND} />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};

export const APIError = Template.bind({});
APIError.args = {};
APIError.parameters = {};
