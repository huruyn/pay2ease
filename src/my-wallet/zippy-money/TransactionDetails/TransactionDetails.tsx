import {
  Button, Col, Row,
} from 'react-bootstrap';
import NumberFormat from 'react-number-format';
import { DateTime } from 'luxon';
import PageContainer from '../../../common/PageContainer';
import { TransactionInterface } from '../../../constants/interface/TransactionInterface';
import { TransactionTypeEnum } from '../../../constants/enum/TransactionTypeEnum';

export default function TransactionDetails({
  user,
  transaction,
  transactionType,
  resetMainInfo,
}: Pick<TransactionInterface, 'user' | 'transaction' | 'transactionType' | 'resetMainInfo'>) {
  const getUserFullName = () => {
    if (transactionType === TransactionTypeEnum.SEND) {
      const contact = transaction?.contact;
      return contact && contact.firstName && contact.lastName ? `${contact.firstName} ${contact.lastName}` : '';
    }
    return user && user.firstName && user.lastName ? `${user.firstName} ${user.lastName}` : '';
  };

  return (
    <>
      <PageContainer
        title="Personal Account"
        subTitle="Made Fun With Zippy!"
        backdropImage="backdrop-image-2"
      >
        <div className="title">
          Transaction Completed
        </div>
        <div className="details">
          <Row>
            <Col xs={6}>To</Col>
            <Col xs={6}>
              <strong>{getUserFullName()}</strong>
            </Col>
          </Row>
          <Row>
            <Col xs={6}>Transfer Date</Col>
            <Col xs={6}>{transaction && transaction.date ? DateTime.fromISO(transaction.date).toUTC().toLocaleString(DateTime.DATE_MED) : ''}</Col>
          </Row>
          <Row>
            <Col xs={6}>Transfer Time (UTC)</Col>
            <Col xs={6}>{transaction && transaction.date ? DateTime.fromISO(transaction.date).toUTC().toLocaleString(DateTime.TIME_WITH_SECONDS) : ''}</Col>
          </Row>
          <Row>
            <Col xs={6}>Transfer Amount</Col>
            <Col xs={6}>
              <NumberFormat
                className="amount"
                value={transaction?.amount}
                defaultValue={0}
                displayType="text"
                prefix="$ "
                suffix=" CAD"
                thousandSeparator
                decimalScale={2}
                fixedDecimalScale
              />
            </Col>
          </Row>
          <Row>
            <Col xs={6}>
              Reference Number
              (Keep For Your Records)
            </Col>
            <Col xs={6}>*3942439898</Col>
          </Row>
          <Row>
            <Col xs={6}>Transaction ID</Col>
            <Col xs={6}>{transaction?.id}</Col>
          </Row>
          <Row>
            <Col xs={6}>Expiry Date</Col>
            <Col xs={6}>{transaction && transaction.expireDate ? DateTime.fromISO(transaction.expireDate).toUTC().toLocaleString(DateTime.DATE_MED) : ''}</Col>
          </Row>
        </div>
        <div className="action">
          <Button
            className="zippy-btn"
            onClick={resetMainInfo}
          >
            Send Another Transfer
          </Button>
        </div>
      </PageContainer>
    </>
  );
}
