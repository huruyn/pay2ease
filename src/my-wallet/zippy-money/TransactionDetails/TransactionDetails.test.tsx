import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TransactionDetails from './TransactionDetails';
import { TransactionTypeEnum } from '../../../constants/enum/TransactionTypeEnum';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useParams: () => ({
    stepId: 'details',
  }),
  useNavigate: () => mockedUsedNavigate,
}));

describe('TransactionDetails Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <TransactionDetails transactionType={TransactionTypeEnum.SEND} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
