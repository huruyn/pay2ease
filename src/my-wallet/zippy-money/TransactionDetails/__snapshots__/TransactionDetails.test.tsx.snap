// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetails Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop backdrop-image-2"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Personal Account
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="title"
        >
          Transaction Completed
        </div>
        <div
          className="details"
        >
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              To
            </div>
            <div
              className="col-6"
            >
              <strong>
                
              </strong>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              Transfer Date
            </div>
            <div
              className="col-6"
            >
              
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              Transfer Time (UTC)
            </div>
            <div
              className="col-6"
            >
              
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              Transfer Amount
            </div>
            <div
              className="col-6"
            >
              <span
                className="amount"
              >
                $ 0.00 CAD
              </span>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              Reference Number (Keep For Your Records)
            </div>
            <div
              className="col-6"
            >
              *3942439898
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              Transaction ID
            </div>
            <div
              className="col-6"
            />
          </div>
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              Expiry Date
            </div>
            <div
              className="col-6"
            >
              
            </div>
          </div>
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            type="button"
          >
            Send Another Transfer
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
